presto-main/src/main/java/com/facebook/presto/operator/aggregation/reservoirsample/UnweightedDoubleReservoirSample.java

```diff
@@ -0,0 +1,171 @@
+        if (other.seenCount < other.samples.length) {
+            for (int i = 0; i < other.seenCount; i++) {
+                add(other.samples[i]);
+            }
+            return;
+        }
+        if (seenCount < samples.length) {
+            final UnweightedDoubleReservoirSample target = ((UnweightedDoubleReservoirSample) other.clone());
+            for (int i = 0; i < seenCount; i++) {
+                target.add(samples[i]);
+            }
+            seenCount = target.seenCount;
+            samples = target.samples;
+            return;
+        }
+
+        shuffleArray(samples);
+        shuffleArray(other.samples);
+        int nextIndex = 0;
+        int otherNextIndex = 0;
+        double[] merged = new double[samples.length];
+        for (int i = 0; i < samples.length; i++) {
+            if (ThreadLocalRandom.current().nextLong(0, seenCount + other.seenCount) < seenCount) {
+                merged[i] = samples[nextIndex++];
+            }
+            else {
+                merged[i] = other.samples[otherNextIndex++];
+            }
+        }
+        seenCount += other.seenCount;
+        samples = merged;
+    }
+
+    @Override
+    public UnweightedDoubleReservoirSample clone()
+    {
+        return new UnweightedDoubleReservoirSample(this);
+    }
+
+    public double[] getSamples()
+    {
+        return Arrays.copyOf(samples, Math.min(seenCount, samples.length));
+    }
+
+    private static void shuffleArray(double[] samples)
+    {
+        for (int i = samples.length - 1; i > 0; i--) {
+            int index = ThreadLocalRandom.current().nextInt(0, i + 1);
+            double sample = samples[index];
+            samples[index] = samples[i];
+            samples[i] = sample;
+        }
+    }
+
+    public static UnweightedDoubleReservoirSample deserialize(SliceInput input)
+    {
+        int seenCount = input.readInt();
+        int maxSamples = input.readInt();
+        double[] samples = new double[maxSamples];
+        input.readBytes(Slices.wrappedDoubleArray(samples), Math.min(seenCount, samples.length) * SizeOf.SIZE_OF_DOUBLE);
+        return new UnweightedDoubleReservoirSample(seenCount, samples);
+    }
+
+    public void serialize(SliceOutput output)
+    {
+        output.appendInt(seenCount);
+        output.appendInt(samples.length);
+        for (int i = 0; i < Math.min(seenCount, samples.length); i++) {
+            output.appendDouble(samples[i]);
+        }
+    }
+
+    public int getRequiredBytesForSerialization()
+    {
+        return SizeOf.SIZE_OF_INT + // seenCount
+                SizeOf.SIZE_OF_INT + // length
+                SizeOf.SIZE_OF_DOUBLE * Math.min(seenCount, samples.length); // samples
+    }
+
+    public long estimatedInMemorySize()
+    {
+        return INSTANCE_SIZE +
+                SizeOf.SIZE_OF_INT + // seenCount
+                SizeOf.SIZE_OF_INT + // length
+                SizeOf.SIZE_OF_DOUBLE * samples.length; // samples
```

Comment:
use sizeOf(samples)

Reply:
I believe I commented on this in one of the last changes: `sizeOf(samples)` is inconsistent with the in-memory version, so I suggest keeping it this way.

Reply:
Searched + replaced in other 3 data structures.